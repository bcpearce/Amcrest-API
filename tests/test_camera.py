"""Tests the camera"""


async def test_serial_number(camera) -> None:
    """Test serial number."""
    assert await camera.async_serial_number == "AMC00123456789ABCDEF"


async def test_lighting(camera, snapshot) -> None:
    """Test lighting."""
    assert await camera.async_lighting_config == snapshot


async def test_get_privacy_mode_on(camera) -> None:
    """Test Privacy mode, fixture was saved with it 'on'."""
    assert await camera.async_get_privacy_mode_on()


async def test_read_physical_config(camera, snapshot) -> None:
    """Test get physical config parameters unexpected to change."""
    assert await camera.async_read_physical_config() == snapshot


async def test_read_ptz_config(camera, snapshot) -> None:
    """Test get PTZ config."""
    assert await camera.async_ptz_preset_info == snapshot
