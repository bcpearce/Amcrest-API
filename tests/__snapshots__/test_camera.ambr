# serializer version: 1
# name: test_lighting
  dict({
    'Lighting': dict({
      0: dict({
        0: dict({
          'Correction': '50',
          'MiddleLight': dict({
            0: dict({
              'Angle': '50',
              'Light': '50',
            }),
          }),
          'Mode': 'Auto',
          'Sensitive': '3',
        }),
        1: dict({
          'Correction': '50',
          'MiddleLight': dict({
            0: dict({
              'Angle': '50',
              'Light': '100',
            }),
          }),
          'Mode': 'Manual',
          'Sensitive': '3',
        }),
        2: dict({
          'Correction': '50',
          'MiddleLight': dict({
            0: dict({
              'Angle': '50',
              'Light': '45',
            }),
          }),
          'Mode': 'Auto',
          'Sensitive': '3',
        }),
        3: dict({
          'Correction': '50',
          'MiddleLight': dict({
            0: dict({
              'Angle': '50',
              'Light': '83',
            }),
          }),
          'Mode': 'Auto',
          'Sensitive': '3',
        }),
      }),
    }),
  })
# ---
# name: test_read_physical_config
  Config(machine_name='AMC0TEST_CAMERA', network={'DefaultInterface': 'eth0', 'Domain': 'dauha', 'Hostname': 'IPC', 'eth0': {'DefaultGateway': '10.0.0.1', 'DhcpEnable': 'true', 'DnsAutoGet': 'false', 'DnsServers': {0: '8.8.8.8', 1: '8.8.4.4'}, 'EnableDhcpReservedIP': 'false', 'IPAddress': 'localhost', 'MTU': '1500', 'PhysicalAddress': 'a0:60:32:ff:ff:ff', 'SubnetMask': '255.0.0.0'}, 'eth2': {'DefaultGateway': '10.0.0.0', 'DhcpEnable': 'true', 'DnsAutoGet': 'false', 'DnsServers': {0: '8.8.8.8', 1: '8.8.4.4'}, 'EnableDhcpReservedIP': 'false', 'IPAddress': '', 'MTU': '1500', 'PhysicalAddress': 'a0:60:32:ff:ff:ff', 'SubnetMask': '255.0.0.0'}}, serial_number='AMC00123456789ABCDEF', session_physical_address='a0:60:32:ff:ff:ff', supported_events=[<EventMessageType.AudioAnomaly: 'AudioAnomaly'>, <EventMessageType.AudioMutation: 'AudioMutation'>, <EventMessageType.LoginFailure: 'LoginFailure'>, <EventMessageType.SafetyAbnormal: 'SafetyAbnormal'>, <EventMessageType.StorageFailure: 'StorageFailure'>, <EventMessageType.StorageLowSpace: 'StorageLowSpace'>, <EventMessageType.StorageNotExist: 'StorageNotExist'>, <EventMessageType.VideoBlind: 'VideoBlind'>, <EventMessageType.VideoMotion: 'VideoMotion'>], software_version='2.800.00AC006.0.R,build:2023-10-27')
# ---
# name: test_read_ptz_config
  list([
    PtzPresetData(index='1', name='Preset1'),
    PtzPresetData(index='2', name='Preset2'),
    PtzPresetData(index='3', name='Preset3'),
    PtzPresetData(index='5', name='Preset5'),
  ])
# ---
